### DISCUSSION TITLE
possibility of defining operators with more than two variables

### DISCUSSION THREAD

**Discussion by Somayeh00** [MilesCranmer/PySR#297] [General]
Dear Miles,
I am new to the Pysr framework and wonder if it is possible to define operators with more than two variables using Pysr. For example, I would like to define some equations that I have shown below. I mean, the entire equation would be a single operator. My ultimate goal is to define a set of basic equations that relate to my process in the code and end up with a physical model.
I thank you in advance for your time and help. 

σ=(2*F)/(π*d_c*h_c )

ρ_c=m_c / ((π/4)*〖d_c〗^2 * h_c )

SSA=(6∑V_i/x_i )/(∑〖ρ_i∙V))

**Comment by MilesCranmer** [comment]
Dear @Somayeh00,

Unfortunately, it is not currently possible to define an operator with more than two inputs. However you might be able to define a custom objective function that performs this, if you want to have this "operator" at the outside of your expressions, with learned inputs for `d_c, h_c, F`, etc. See, e.g., https://github.com/MilesCranmer/PySR/discussions/299 and https://github.com/MilesCranmer/PySR/issues/285 for more info.

In the future I want to allow for fixed functional forms. See https://github.com/MilesCranmer/SymbolicRegression.jl/issues/193 for updates.

Cheers,
Miles

**Reply by Somayeh00** [reply]
Thanks for your support. I'll look into defining a custom objective function, as you suggested.
